src/utils/hdnode.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * @packageDocumentation
+ * @module Utils-HDNode
+ */
+
+import { Buffer } from 'buffer/'
+import hdnode from 'hdkey'
+
+/**
+ * BIP32 hierarchical deterministic keys.
+ */
+
+export default class HDNode {
+  private hdkey: any
+  publicKey: Buffer
+  privateKey: Buffer
+  chainCode: Buffer
+  privateExtendedKey: string
+  publicExtendedKey: string
+
+  /**
+  * Derives the HDNode at path from the current HDNode.
+  * @param path 
+  */
+  derive(path: string): HDNode {
+    const hdKey = this.hdkey.derive(path)
+    let hdNode: HDNode
+    if (hdKey.privateExtendedKey != null) {
+      hdNode = new HDNode(hdKey.privateExtendedKey)
+    } else {
+      hdNode = new HDNode(hdKey.publicExtendedKey)
+    }
+    return hdNode
+  }
+
+  /**
+  * Signs the buffer hash with the private key using secp256k1 and returns the signature as a buffer.
+  * @param hash 
+  */
+  sign(hash: Buffer): Buffer {
+    const sig: Buffer = this.hdkey.sign(hash)
+    return Buffer.from(sig)
+  }
+
+  /**
+  * Verifies that the signature is valid for hash and the HDNode's public key using secp256k1.
+  * @param hash 
+  * @param signature 
+  * @returns true for valid, false for invalid. Throws if the hash or signature is the wrong length.
```

Comment:
Similarly, would an `@throws` be appropriate?

Reply:
Agreed. Resolved.